accepted/2022/simplify-output-paths.md

```diff
@@ -0,0 +1,58 @@
+
+Currently, the default output path for a .NET project includes folders for the `TargetFramework` and the `RuntimeIdentifier` (if it is set).  Additionally, the publish output goes in a `publish` subfolder of the output folder.  The resulting paths are as follows:
+
+- `bin\<Configuration>\<TargetFramework>` - Build output with no `RuntimeIdentifier`
+- `bin\<Configuration>\<TargetFramework>\<RuntimeIdentifier>` - Build output with `RuntimeIdentifier`
+- `bin\<Configuration>\<TargetFramework>\publish` - Publish output with no `RuntimeIdentifier`
+- `bin\<Configuration>\<TargetFramework>\<RuntimeIdentifier>\publish` - Publish output with `RuntimeIdentifier`
+
+This is rather complicated an inconsistent, which can make for a poor first impression of the .NET platform.
+
+We'd like to try to improve the output path structure.  Desired qualities include:
+
+- Simple and consistent
+- Avoid cases where one output path is nested inside of another one
+- Avoid excessively deep directory heirarchies
+- Get rid of `obj` folder in project root
+
+## Proposed behavior
+
+Projects targeting .NET and higher will by default use a new output path format.  The output path will consist of the following 3 nested folders
+
+- `bin` - All output (including intermediate output, will go under this folder)
+- Output Type - Such as `build`, `publish`, `obj`, or `packages`
+- Pivots - This will at minimum be the `Configuration`, such as `Debug` or `Release`. Other pivots such as `TargetFramework` or `RuntimeIdentifier` may also be included, and the pivots will be joined by the underscore (`_`) character
+  - `TargetFramework` will be included in the folder name if the project is multi-targeted (`TargetFrameworks` is non-empty), or if the `TargetFramework` property was set via the command line (ie is a global property)
+  - `RuntimeIdentifier` will be included in the folder name if it was explicitly set (either in a project file or on the command line).  If it is set automatically by the SDK (for exmaple because `SelfContained` was set)
+
+Some examples:
+
+- `bin\build\Debug` - The build output path for a simple project when you run `dotnet build`
+- `bin\obj\Debug` - The intermediate output path for a simple project when you run `dotnet build`
+- `bin\build\Debug_net8.0` - The build output path for the `net8.0` build of a multi-targeted project
+- `bin\publish\Release_linux-x64` - The publish path for a simple app when publishing for `linux-x64`
+- `bin\package\Release` - The folder where the release .nupkg will be created for a project
+
+## Considerations
+
+### Breaking changes
+
+These changes could break things such as scripts that copy the output of the build or custom MSBuild logic that hard-codes these paths.  Tieing these changes to the project TargetFramework ensures that these breaks will be encountered when the project is modified to target a new TargetFramework, not when updating to a new version of the .NET SDK.
```

Comment:
just to confirm: if I multi-target to both net7.0 and net8.0 I get both folder layouts right?